src/main/java/com/cloudogu/scm/repositorytemplate/RepositoryTemplater.java

```diff
@@ -0,0 +1,187 @@
+      for (FileObject child : file.getChildren()) {
+        BrowserResult result = templateService.getBrowseCommand().setPath(child.getPath()).setRecursive(true).getBrowserResult();
+        copyFileRecursively(templateService, filter, templateFile, result.getFile());
+      }
+    } else {
+      copyToTargetRepository(templateService, filter, templateFile, file.getPath());
+    }
+  }
+
+  private void copyToTargetRepository(RepositoryService templateService, TemplateFilter filter, RepositoryTemplateFile templateFile, String filePath) throws IOException {
+    try (InputStream content = templateService.getCatCommand().getStream(filePath)) {
+      if (templateFile.isFiltered()) {
+        copyFileTemplated(filter, content, filePath);
+      } else {
+        copyFile(content, filePath);
+      }
+    }
+  }
+
+  private void copyFileTemplated(TemplateFilter templateFilter, InputStream content, String filepath) throws IOException {
+    InputStream templatedContent = new ByteArrayInputStream(templateFilter.filter(content, filepath).toByteArray());
+    context.create(filepath).from(templatedContent);
+  }
+
+  private void copyFile(InputStream content, String filepath) throws IOException {
+    context.create(filepath).from(content);
+  }
+
+  private TemplateEngine setupTemplateEngine(RepositoryTemplate repositoryTemplate) {
+    String configEngine = repositoryTemplate.getEngine() != null ? repositoryTemplate.getEngine() : "mustache";
+    TemplateEngine engine = engineFactory.getEngine(configEngine);
+    if (engine == null) {
+      throw NotFoundException.notFound(ContextEntry.ContextBuilder.entity(TemplateEngine.class, configEngine));
+    }
+    return engine;
+  }
+
+  private RepositoryTemplate parseTemplateConfig(RepositoryService repositoryService) throws IOException {
+    Optional<String> templateFile = RepositoryTemplateFinder.templateFileExists(repositoryService);
+    if (!templateFile.isPresent()) {
+      throw NotFoundException.notFound(
+        ContextEntry.ContextBuilder.entity(RepositoryTemplate.class, "template file").in(repositoryService.getRepository())
+      );
+    }
+    try (InputStream templateYml = repositoryService.getCatCommand().getStream(templateFile.get())) {
+      RepositoryTemplate repositoryTemplate = unmarshallRepositoryTemplate(templateFile.get(), templateYml);
+      repositoryTemplate.setNamespaceAndName(repositoryService.getRepository().getNamespaceAndName().toString());
+      if (repositoryTemplate.getFiles() == null) {
+        repositoryTemplate.setFiles(Collections.emptyList());
+      }
+      return repositoryTemplate;
+    }
+  }
+
+  private RepositoryTemplate unmarshallRepositoryTemplate(String templateFile, InputStream templateYml) {
+    RepositoryTemplate repositoryTemplate = yaml.loadAs(templateYml, RepositoryTemplate.class);
+    if (repositoryTemplate == null) {
+      throw new YAMLException("repository template invalid -> could not parse " + templateFile);
+    }
```

Comment:
A failed parsing leads to an internal server error in the ui. This should be a backend error which describes the problem.